### ISSUE TITLE
Editor does not seem to recgonize Java types usually found in the default JRE

### ISSUE THREAD

**Issue by FaceFTW** [codenvy/codenvy#1461]
The Editor seems to mark types that are found in the JRE as errors saying "<Insert Type Here> cannot be resolved"
**Reproduction Steps:**
Imported a project from Github
Reconfigured Source and Library paths
Opened a file
Errors appeared (Exact Message: "ArrayList cannot be resolved to a type")

**Expected behavior:**

No errors should be existing saying that types cannot be resolved

**Observed behavior:**

The external library types were not affected, but some of the JRE types, such as ArrayList are marked saying it cannot be resolved

**Codenvy version:** 5.0.0-M8
**OS and version:** Windows 10 (Computer)
**Browser version:** Microsoft Edge

**Other questions:**
Problem happens at codenvy.io
Problem can be reliably reproduced:  it's random

**Comment by FaceFTW** [comment]
Before I forget, it does not have any problems with primitive types (ie. int, long, double, char, etc.), but other non-primitives I found it marks are String, FileBuffer, FileWriter, and the Override annotation. It also does not seem to read the abstract modifier, and asks for an explicit constructor in an abstract class

**Comment by TylerJewell** [comment]
As a diagnostic / investigation activity - can you confirm that this issue happens for both "javac" project types and "maven" project types with a simple pom.xml in the project root? We spend most of our time implementing Che with a maven project type in Che, so we would see these errors appear, and generally haven't yet.

BTW - these are Eclipse Che issues, for which we are built upon - more than a Codenvy specific issue.

**Comment by FaceFTW** [comment]
Sure! I can probably get you results by tomorrow. Also, I am using Ant to compile my project, as I am also using Eclipse Neon.2 when I can

**Comment by TylerJewell** [comment]
Yeah, even with Ant, you could still do a simple maven project type with a dumb pom.xml in the root. Our maven PT has a specialized dependency analyzer and downloader, so I am wondering if our language analyzers is reacting differently for that vs. the javac type.

**Comment by FaceFTW** [comment]
No Problem, I may have figured out some of the issue, because I had to install ant again, but the maven build is failing, so I am trying to fix that

**Comment by FaceFTW** [comment]
I think I may have fixed the problem. The problem may have been that when I imported from Github, the original Eclipse project configuration was looking for JDK 1.8, which was not installed on the actual machine. This may explain why the editor was marking those types, because it could not find the default JRE 1.8 Libraries where it should be. I am not completely sure, but when I originally tried to run ant, it could not find a valid Java Compiler, so I installed OpenJDK8, and now there seems to be no problem.
